### PR TITLE
refactor(backup): move collect_backup_info to replica_backup_manager

### PR DESCRIPTION
**Pull request by levy5307**
move collect_backup_info from replica to replica_backup_manager

### CODE REVIEW

File: src/dist/replication/lib/backup/replica_backup_manager.cpp [hunk 1]
```diff
@@ -44,6 +44,15 @@ static bool get_policy_checkpoint_dirs(const std::string &dir,
     return true;
 }
 
+replica_backup_manager::replica_backup_manager(replica *r) : replica_base(r), _replica(r) {}
+
+replica_backup_manager::~replica_backup_manager()
+{
+    if (_collect_info_timer != nullptr) {
+        _collect_info_timer->cancel(false);
```
Comment (neverchanje) [inline src/dist/replication/lib/backup/replica_backup_manager.cpp:52]:
```suggestion
        _collect_info_timer->cancel(true);
```
Why change the true here to false?

File: src/dist/replication/lib/backup/replica_backup_manager.h [hunk 2]
```diff
@@ -12,15 +12,20 @@ namespace replication {
 class replica_backup_manager : replica_base
 {
 public:
-    replica_backup_manager(replica *r) : replica_base(r), _replica(r) {}
+    replica_backup_manager(replica *r);
```
Comment (neverchanje) [inline src/dist/replication/lib/backup/replica_backup_manager.h:15]:
```suggestion
    explicit replica_backup_manager(replica *r);
```

File: src/dist/replication/lib/backup/replica_backup_manager.cpp [hunk 3]
```diff
@@ -73,6 +82,51 @@ void replica_backup_manager::on_clear_cold_backup(const backup_clear_request &re
     background_clear_backup_checkpoint(request.policy_name);
 }
 
+void replica_backup_manager::start_collect_backup_info()
+{
+    _collect_info_timer =
```
Comment (neverchanje) [inline src/dist/replication/lib/backup/replica_backup_manager.cpp:87]:
```suggestion
  if(_collect_info_timer == nullptr) {
    _collect_info_timer =
  }
```

File: src/dist/replication/lib/backup/replica_backup_manager.cpp [hunk 4]
```diff
@@ -44,6 +44,15 @@ static bool get_policy_checkpoint_dirs(const std::string &dir,
     return true;
 }
 
+replica_backup_manager::replica_backup_manager(replica *r) : replica_base(r), _replica(r) {}
+
+replica_backup_manager::~replica_backup_manager()
+{
+    if (_collect_info_timer != nullptr) {
+        _collect_info_timer->cancel(true);
```
Comment (hycdong) [inline src/dist/replication/lib/backup/replica_backup_manager.cpp:52]: Will `cancel` timer guarantee `_collect_info_timer = nullptr`, so you remove it?

Comment (levy5307) [reply src/dist/replication/lib/backup/replica_backup_manager.cpp:52]: the destructor of replica_backup_manager will guarantee  `_collect_info_timer = nullptr`, not cancel

**Pull request merged**